onnxruntime/core/providers/xnnpack/detail/utils.h

```diff
@@ -56,6 +57,50 @@ enum class QuantizedOpType : uint8_t {
   Unknown,
 };
 
+class InternalDataInitializer {
+ public:
+  InternalDataInitializer(const ONNX_NAMESPACE::TensorProto& tensor_proto);
+
+  bool HasExternalData() {
+    return has_external_data_;
+  }
+  const TensorShape& Shape() {
+    return shape_;
+  }
+
+  const DataTypeImpl* Type() {
+    return type_;
+  }
+
+  const void* raw_data() {
+    return unpacked_tensor_.data();
+  }
+
+  template <class T>
+  gsl::span<const T> DataSpan() {
+    return gsl::make_span(data<T>(), unpacked_tensor_.size() / sizeof(T));
+  }
+
+  template <class T>
+  const T* data() {
```

Comment:
I would make this private so users can only get the data in a span.

Reply:
Good suggestion,  expose `DataSpan`  as public is enough